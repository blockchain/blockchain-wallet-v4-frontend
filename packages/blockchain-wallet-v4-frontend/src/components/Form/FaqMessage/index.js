import React from 'react'
import styled from 'styled-components'
import { Text, Icon } from 'blockchain-info-components'

const Container = styled.div`
  box-sizing: border-box;
  border: 1px solid ${props => props.theme['gray-2']};
  padding: 20px;
  max-width: 300px;
`
const Title = styled(Text)`
  line-height: 20px;
  font-size: 14px;
  font-weight: 600;
  margin-bottom: 10px;
`
const Message = styled(Text)`
  font-size: 14px;
  line-height: 20px;
  font-weight: 300;
`
const FaqIcon = styled(Icon)`
  color: ${props => props.theme['brand-primary']};
  font-size: 24px;
  display: flex;
  width: 24px;
  justify-content: center;
  height: 24px;
  line-height: 24px;
  font-weight: 400;
  margin-bottom: 16px;
`

const FaqMessage = ({ title, text, icon, ...rest }) => (
  <Container {...rest}>
    <FaqIcon name={icon} />
    <Title>{title}</Title>
    <Message>{text}</Message>
  </Container>
)

export default FaqMessage
