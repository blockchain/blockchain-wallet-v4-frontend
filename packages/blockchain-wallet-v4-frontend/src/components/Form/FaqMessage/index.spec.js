import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import FaqMessage from './index.js'

describe('FooterShadowWrapper', () => {
  it('renders correctly', () => {
    const props = { title: <div />, text: <div />, className: 'cn' }
    const component = shallow(<FaqMessage {...props} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
