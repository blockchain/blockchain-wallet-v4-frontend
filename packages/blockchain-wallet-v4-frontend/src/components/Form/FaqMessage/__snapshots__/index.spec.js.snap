// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterShadowWrapper renders correctly 1`] = `
<FaqMessage__Container
  className="cn"
>
  <FaqMessage__FaqIcon />
  <FaqMessage__Title>
    <div />
  </FaqMessage__Title>
  <FaqMessage__Message>
    <div />
  </FaqMessage__Message>
</FaqMessage__Container>
`;
